### DISCUSSION TITLE
Support for multi-schema/multi-graphql-endpoint projects (with pre-existing solution)

### DISCUSSION THREAD

**Discussion by ZMobile** [Netflix/dgs-framework#220] [Ideas]
Until recently, GraphQL allowing for only one schema/endpoint per project was seen as a big but unavoidable limitation.

Then about a year ago, Github user Jim Kynde Meyer perfected his js-graphql-intellij-plugin allowing for graphql-java projects to have multiple schemas, multiple graphql endpoints, and even schemas shared by other schemas. With this plugin, the schemas are mapped & distinguished with either one (or one per schema) .graphqlconfig file. As quoted by Jim: "GraphQL files find their associated .graphqlconfig by looking in their current and parent directories."

Here is a link to the plugin:
https://github.com/jimkyndemeyer/js-graphql-intellij-plugin

Here's one of his examples:
https://github.com/jimkyndemeyer/graphql-config-examples/tree/master/two-schemas-plus-shared-using-projects

I'm very impressed with your DGS framework and pleased you made it open source. It singlehandedly solves every reservation I had about using graphQL with Spring Boot. Player from Carmen Sandiego would be proud.

There is only one single major issue specifically with multi-endpoint GraphQL projects: it seems to butt heads with the syntax of my .graphqlconfig files and won't run unless they're gone. However in some cases, those .graphqlconfig files still have to be present or else the multi endpoint structure breaks down, so I end up having to relocate the .graphqlconfig mapping to a single file in the root directory where it's out of range of netflix-dgs in order for the project to function.

This works, but I wish it wouldn't clash with the .graphqlconfig files in the first place, as it would in turn mean full support for graphQL projects with multiple schemas & endpoints.

Here is a copy of the error that will be produced when you try to run a spring boot dgs application within range of a .graphqlconfig file: 
`
There are problems with the GraphQL Schema:
	 * InvalidSyntaxError{ message=Invalid Syntax : offending token '"name"' at line 2 column 3 ,offendingToken="name" ,locations=[SourceLocation{line=2, column=3}] ,sourcePreview={
  "name": "GraphQL Config",
`

**Comment by paulbakker** [comment]
Can you share a sample `.graphqlconfig` that reproduces the error? I'm not quite sure in what scenario it breaks.

**Reply by ZMobile** [reply]
Absolutely. Here is a sample of a simple .graphqlconfig file for a single schema:
`
{
  "name": "Example Schema",
  "schemaPath": "schema.graphql",
  "extensions": {
    "endpoints": {
      "Example GraphQL Endpoint": {
        "url": "http://localhost:8000/graphql",
        "headers": {
          "user-agent": "JS GraphQL"
        },
        "introspect": false
      }
    }
  }
}
`

To be organized, I’ll usually have it located either in the same package as it’s corresponding schema.graphql file (in Netflix dgs’ case that’d be resources.schema file), or right next to the package that contains it. However, Netflix dgs will not like it, and throw the same error no matter what it’s contents are, even if the .graphqlconfig JSON is just an empty pair of braces like so:
`
{}
`
